server/controllers.js

```diff
@@ -44,6 +66,248 @@ const controllers = {
+
+  addTaskHomeAddress: (req, res) => {
+    const { id } = req.params;
+
+    const {
+      description,
+      carRequired,
+      laborRequired,
+      category,
+      date,
+      time,
+      dateRequested,
+      timeRequested,
+      duration,
+    } = req.body;
+
+    const queryStr = `
+      INSERT INTO nexdoor.tasks (
+        requester_id,
+        location_id,
+        description,
+        car_required,
+        physical_labor_required,
+        status,
+        category,
+        date,
+        time,
+        date_requested,
+        time_requested,
+        duration
+      ) VALUES (
+        ${id},
+        (
+          SELECT address_id
+          FROM nexdoor.users
+          WHERE user_id=${id}
+        ),
+        '${description}',
+        ${carRequired},
+        ${laborRequired},
+        'open', '${category}',
+        '${date}',
+        '${time}',
+        '${dateRequested}',
+        '${timeRequested}',
+        ${duration}
+      );
+    `;
+
+    db.query(queryStr, (err, data) => {
+      if (err) {
+        res.status(400).send(err.stack);
+      } else {
+        res.status(200).send(data.rows);
+      }
+    });
+  },
+
+  // ******************************************************
+  addAnnouncement: (req, res) => {
+    const { userId } = req.params;
+    const {
+      announcementBody,
+      date,
+      time,
+    } = req.body;
+
+    const queryStr = `
+      INSERT INTO nexdoor.announcements (
+        user_id,
+        announcement_body,
+        date,
+        time
+      )
+      VALUES (
+        ${userId},
+        ${announcementBody},
+        ${date},
+        ${time}
+      )
+    `;
+
+    db.query(queryStr, (err, data) => {
+      if (err) {
+        res.status(400).send(err.stack);
+      } else {
+        res.status(200).send(data.rows);
+      }
+    });
+  },
+
+  getUser: (req, res) => {
+    const { id } = req.params;
+
+    const queryStr = `
+      SELECT
+        firstname,
+        lastname,
+        email,
+        karma,
+        task_count,
+        avg_rating,
+        profile_picture_url, (
+          SELECT JSON_AGG(ROW_TO_JSON(t))
+          FROM (
+            SELECT
+              street_address,
+              city,
+              state,
+              zipcode,
+              neighborhood
+            FROM nexdoor.address
+            WHERE address_id=nexdoor.users.address_id
+          ) t
+        ) as address
+      FROM nexdoor.users
+      WHERE user_id=${id};
+    `;
+
+    db.query(queryStr, (err, data) => {
+      if (err) {
+        res.status(400).send(err.stack);
+      } else {
+        res.status(200).send(data.rows);
+      }
+    });
+  },
+
+  checkForEmail: (req, res) => {
+    const { email } = req.body;
+    const queryStr = `
+      SELECT EXISTS (
+        SELECT true FROM nexdoor.users
+        WHERE email='${email}'
+        LIMIT 1
+      )
+    `;
+    db.query(queryStr, (err, data) => {
+      if (err) {
+        res.status(400).send(err.stack);
```

Comment:
Can you switch the error block around so that it throws an error if the email address exists in the database?

Reply:
actually nevermind, I'll do error handling in validation